File: step03view/src/main/java/com/example/step03view/MainActivity.java
package com.example.step03view;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    //필드
    private EditText inputMsg;
    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // res/layout/activity_main.xml 문서를 전개해서 화면 구성하기
        setContentView(R.layout.activity_main);

        /*
            위의 xml 문서를 전개하면
            ConstraintLayout, EditText, Button 객체가 생성된다.
            만일 java code에서 해당 UI를 가지고 어떤 동작을 하려면 그 객체의 참조값이 필요하다
            그 객체의 참조값을 어떻게 얻어오는지를 학습하기
         */

        //id를 이용해서 EditText 객체의 참조값 얻어오기
        this.inputMsg=findViewById(R.id.inputMsg);

        //id를 이용해서 Button 객체의 참조값 얻어오기
        Button sendBtn = findViewById(R.id.sendBtn);

        //버튼을 클릭했을 때 호출될 메소드를 가지고 있는 View.OnClickListener type의 참조값 전달하기
        sendBtn.setOnClickListener(this);

        //id를 이용해서 TextView 객체의 참조값 얻어오기
        this.textView = findViewById(R.id.textView);

    }

    //버튼을 클릭하면 호출되는 메소드
    @Override
    public void onClick(View view) {
        //EditText에 입력한 문자열을 읽어와서 지역변수에 담기
        String msg = this.inputMsg.getText().toString();

        textView.setText(msg);
        //만약 숫자를 출력하고 싶으면 아래와 같은 형태로 출력을 해야한다(문자열 타입으로 변환시켜야함)
        // 그냥 숫자를 넣을경우(setText(100)), 아이디값으로 알아듣기 때문이다 .
        // this.textView.setText(Integer.toString(100));
        
        //빈 문자열을 입력해서 입럭창 지우기
        inputMsg.setText("");

        Toast.makeText(this,msg,Toast.LENGTH_SHORT).show();
    }
}

File: step17httprequest3/src/main/java/com/example/step17httprequest3/MainActivity.kt
package com.example.step17httprequest3

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import org.json.JSONArray
import org.json.JSONObject

class MainActivity : AppCompatActivity() , Util.RequestListener{
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //EditText 객체의 참조값 얻어오기
        val editText=findViewById<EditText>(R.id.inputMsg)
        //Button 객체의 참조값 얻어와서 리스너 등록
        val sendBtn=findViewById<Button>(R.id.sendBtn)
        sendBtn.setOnClickListener{
            //EditText 에 입력한 문자열을 읽어와서
            val msg=editText.text.toString()
            Util.sendPostRequest(
                    999,
                    "http://192.168.0.31:9000/boot07/api/send",
                    mapOf("msg" to msg),  // "msg" 라는 키값으로 입력한 메세지를 담은 Map
                    this
            )
        }
        val getListBtn=findViewById<Button>(R.id.getListBtn);
        getListBtn.setOnClickListener{
            Util.sendGetRequest(
                    1000,
                    "http://192.168.0.31:9000/boot07/api/list",
                    mapOf("pageNum" to "1"),
                    this
            )
        }
    }

    override fun onSuccess(requestId: Int, result: Map<String, Any?>?) {
        if(requestId == 999){
            //웹서버가 응답한 json 문자열 { }
            val jsonStr = result?.get("data").toString()
            Log.d("#### json 문자열 ####", jsonStr)
            // JSONObject 객체를 생성하면서 생성자의 인자로 json 문자열을 전달한다.
            val obj=JSONObject(jsonStr)
            // key 값을 이용해서 Boolean, String, Int 값을 추출할수 있다.
            val isSuccess=obj.getBoolean("isSuccess")
            val response=obj.getString("response")
            val num=obj.getInt("num")
        }else if(requestId == 1000){
            val jsonStr = result?.get("data").toString();
            Log.d("#### json 문자열 ####", jsonStr)
            val arr=JSONArray(jsonStr)
            //반복문 돌면서 i 값을 0 에서 부터 JSONArray 의 방의 사이즈 - 1 까지 변화 시킨다
            for(i in 0..arr.length()-1){
                val tmp = arr.getString(i)
                Log.d("json array", tmp)
            }
        }
    }

    override fun onFail(requestId: Int, result: Map<String, Any?>?) {

    }
}